packages/cdk-assets/lib/private/docker.ts

```diff
@@ -70,9 +86,49 @@ export class Docker {
     await this.execute(['push', tag]);
   }
 
+  /**
+   * If a CDK Docker Credentials file exists, creates a new Docker config directory.
+   * Sets up `docker-credential-cdk-assets` to be the credential helper for each domain in the CDK config.
+   * All future commands (e.g., `build`, `push`) will use this config.
+   *
+   * See https://docs.docker.com/engine/reference/commandline/login/#credential-helpers for more details on cred helpers.
+   *
+   * @returns true if CDK config was found and configured, false otherwise
+   */
+  public configureCdkCredentials(): boolean {
+    const config = cdkCredentialsConfig();
+    if (!config) { return false; }
+
+    const configDir = this.createCleanConfig();
+
+    const domains = Object.keys(config.domainCredentials);
+    const credHelpers = domains.reduce(function(map: Record<string, string>, domain) {
+      map[domain] = 'cdk-assets'; // Use docker-credential-cdk-assets for this domain
```

Comment:
Are we sure the tools is going to be on the `$PATH`? Do we need to do something to ensure that?

Reply:
For the initial target of this (pipelines), the `cdk-assets` is installed globally, so it's on the path. On local machines, this can be reasonably easily set up as well. Was also going to vote "punt" on this until we had more concrete requirements.